pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+            logger.debug(
+                "Missing pk_field, creating auto-id pk for collection: %s",
+                self.collection_name,
+            )
+        else:
+            # If pk_field given, assume it was iterated
+            try:
+                fields[self.pk_field]["auto_id"] = False
+                fields[self.pk_field]["is_primary"] = True
+            except KeyError as ex:
+                logger.error(
+                    "Missing pk_field: %s  in data for collection: %s",
+                    self.pk_field,
+                    self.collection_name,
+                )
+                raise ex
+        try:
+            # Create the fieldschemas
+            fieldschemas = []
+            # TODO: Assuming ordered dicts for 3.7
+            self.fields = {}
+            for field_dict in fields.values():
+                fieldschemas.append(FieldSchema(**field_dict))
+                self.fields[field_dict["name"]] = field_dict["dtype"]
+            # Create the schema for the collection
+            schema = CollectionSchema(fieldschemas)
+            # Create the collection
+            self.collection = Collection(
+                name=self.collection_name,
+                schema=schema,
+                consistency_level=self.consistency_level,
+                shards_num=self.shard_num,
+                using=self.alias,
+            )
+            logger.error("Successfully created collection: %s", self.collection_name)
+        except MilvusException as ex:
+            logger.error("Failed to create collection: %s", self.collection_name)
+            raise ex
+
+    def _extract_fields(self) -> None:
+        """Grab the existing fields from the Collection"""
+        self.fields = {}
+        schema = self.collection.schema
+        for field in schema.fields:
+            field_dict = field.to_dict()
+            if field_dict.get("is_primary", None) is not None:
+                logger.debug("Updating pk_field with one from collection.")
+                self.pk_field = field_dict["name"]
+            if field_dict["type"] in (DataType.FLOAT_VECTOR, DataType.BINARY_VECTOR):
+                logger.debug("Updating vector_field  with one from collection.")
+                self.vector_field = field_dict["name"]
+            self.fields[field_dict["name"]] = field_dict["type"]
+
+        logger.info(
+            "Successfully extracted fields from for collection: %s, total fields: %s, "
+            "pk_field: %s, vector_field: %s",
+            self.collection_name,
+            len(self.fields),
+            self.pk_field,
+            self.vector_field,
+        )
+
+    def _create_index(self) -> None:
+        """Create a index on the collection"""
+        if self._get_index() is None:
+            # If no index params, use a default HNSW based one
+            if self.index_params is None:
+                # TODO: Once segment normalization we can default to IP
+                metric_type = (
```

Comment:
user has to be able to set default metric type I guess

Reply:
we are gonna to support cosine soon,so far the default set to L2 works for me but it has to been changed soon.

Reply:
The user is able to set an index at init if they want to use their own index params.